internal/config/config.go

```diff
@@ -36,6 +36,7 @@ type Config struct {
 	UseDualStackEndpoint           bool
 	UseFIPSEndpoint                bool
 	UserAgent                      UserAgentProducts
+	WebIdentityToken               string
```

Comment:
The `WebIdentityToken` should be part of the block used to assume the role, so that it's all defined in one place. This will be more important when, e.g. https://github.com/hashicorp/terraform-provider-aws/issues/22728 is implemented.

Reply:
We may also want to allow specifying the token file, in addition to putting it inline

Reply:
`WebIdentityToken` is not part of `AssumeRoleWithWebIdentity`. I also added option to pass `WebIdentityTokenFile`.